File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
        for(char c : s.toCharArray()) {
            if(c == '(') {st.push(')');}
            else if(c == '[') {st.push(']');}
            else if(c == '{') {st.push('}');}
            else if(st.empty() || st.peek() != c) {return false;}
            else {st.pop();}
        }
        return st.empty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char> st = new Stack<char>();
        foreach (char c in s)
        {
            if (c == '(') { st.Push(')'); }
            else if (c == '[') { st.Push(']'); }
            else if (c == '{') { st.Push('}'); }
            else if (st.Count == 0 || st.Peek() != c) { return false; }
            else { st.Pop(); }
        }
        return st.Count == 0;
    }
}